Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -205,7 +205,7 @@ private static void HandleSessionError(IntPtr sessionHandlePtr, ErrorCode errorC
         {
             try
             {
-                var handle = new SessionHandle(sessionHandlePtr);
+                using var handle = new SessionHandle(sessionHandlePtr);
```

Comment:
This was causing issues since it was extending the lifetime of the session in the event of an error, thus preventing the Realm from being deleted. It has always been a problem, but the Core changes exposed altered the timing, exposing it in our unit tests.